### PR TITLE
[ParseableInterface] Distinguish SDK and non-SDK dependencies

### PR DESCRIPTION
**Pull request by jrose-apple**
Honor -track-system-dependencies when building swiftinterfaces into swiftmodules on demand, then start distinguishing SDK and non-SDK dependencies to allow for relocating SDKs.

One interesting thing here is that the decision of whether or not to track system dependencies ends up going into the cache key for a swiftmodule built from a parseable interface, because it affects that module's up-to-date check. If we didn't include `-track-system-dependencies` in the cache key, we could end up tracking system dependencies for some modules but not others in the same build.

There's a bit of a bug here where `-track-system-dependencies` is *not* honored if the top-level invocation isn't tracking *any* dependencies, but given how uncommon this flag and that situation are in practice that's probably okay for now.

Still TODO: honor this for `-build-swiftmodule-from-parseable-interface` as well. That's currently not tracking dependencies at all and it probably should.

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

File: lib/Frontend/ParseableInterfaceSupport.cpp [hunk 1]
```diff
@@ -307,7 +324,25 @@ collectDepsForSerialization(clang::vfs::FileSystem &FS,
       return true;
     uint64_t mtime =
       Status->getLastModificationTime().time_since_epoch().count();
-    Deps.push_back(FileDependency{Status->getSize(), mtime, DepName});
+
+    bool IsSDKRelative = false;
+    if (SDKPath.size() > 1 && DepName.startswith(SDKPath)) {
+      assert(DepName.size() > SDKPath.size() &&
+             "should never depend on a directory");
+      if (llvm::sys::path::is_separator(DepName[SDKPath.size()])) {
```
Comment (harlanhaskins) [inline lib/Frontend/ParseableInterfaceSupport.cpp]: Can you add a comment clarifying what this line is looking for?

**Review by harlanhaskins — APPROVED** [review]

**Comment by harlanhaskins** [comment]
Can you also update the comment in ModuleLoader.cpp that says we currently don't pass `true` for system dependencies?

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
The tests probably need to be updated.

**Comment by jrose-apple** [comment]
@nathawes picked this up in #23665.

**Pull request closed without merging**